Dockerfile

```diff
@@ -106,11 +106,26 @@ CMD ["sh", "-c", "/bin/barbican-kms-plugin --socketpath ${socketpath} --cloud-co
 ##
 ## cinder-csi-plugin
 ##
-FROM --platform=${TARGETPLATFORM} ${DEBIAN_IMAGE} as cinder-csi-plugin
+FROM --platform=${TARGETPLATFORM} ${DEBIAN_IMAGE} as cinder-csi-plugin-utils
 
-# Install e4fsprogs for format
-RUN clean-install btrfs-progs e2fsprogs mount udev xfsprogs
+RUN clean-install bash rsync mount udev btrfs-progs e2fsprogs xfsprogs
+COPY tools/csi-deps.sh /tools/csi-deps.sh
+RUN /tools/csi-deps.sh
 
+##
+FROM --platform=${TARGETPLATFORM} ${DISTROLESS_IMAGE} as cinder-csi-plugin-utils-check
+
+COPY --from=cinder-csi-plugin-utils /bin/sh /bin/sh
+COPY --from=cinder-csi-plugin-utils /dest /
+COPY tools/csi-deps-check.sh /tools/csi-deps-check.sh
+
+SHELL ["/bin/sh"]
+RUN /tools/csi-deps-check.sh
```

Comment:
This looks weird to the casual fly-by maintainer because we're not actually using it. It's purpose is, I assume, to cause an error into the build process if csi-deps.sh didn't do its job properly?

Assuming I've got that right, please could you add the following comments:

* At the top of these section explaining that this is simply a sanity check on the previous build step and its output is not used.
* In csi-deps.sh explaining that any changes there should be reflected in csi-deps-check.sh

Reply:
Thank you for advice. aseptically "future me" )

I've added a few comments to the shell file and Dockerfile + small refactoring.